### ISSUE TITLE
'Import objects from Rockstar' feature

### ISSUE THREAD

**Issue by blattersturm** [citizenfx/fivem#873]
A little silly idea, but imagine if the world editor had an option to let you paste in a Rockstar job URL or ID and it'd (try to) fetch the JSON file underlying the job and place all objects (and later on metadata like enemies, pickups, spawns, or checkpoints?) in your current map - a little cute way to get a lot of examples to remix as well.

I might be able to pick this idea up myself if nobody else will.

**Comment by jaymo1011** [comment]
Those JSON files are **not** fun to parse. Perhaps there could be a community effort to document how the freemode mission controller parses the JSON? (Or if someone already has this documented, ask them to share?)

**Comment by sadboilogan** [comment]
> 
> 
> Those JSON files are **not** fun to parse. Perhaps there could be a community effort to document how the freemode mission controller parses the JSON? (Or if someone already has this documented, ask them to share?)

agreed, would definitely be helpful for not only this but for other cases where UGC content can be parsed and used for some interesting gamemodes and such

**Comment by RussianRonin** [comment]
This is a parser that a friend of mine developed for parsing the json data from the image url of the job. It only grabs race related information (checkpoints, props) as it is a racing resource: https://github.com/mkafrin/mkr-racing/blob/nopixel/raceparser/RaceParser.js#L78

**Comment by blattersturm** [comment]
Your link seems to lead to a 404. However, there's _already_ various implementations of this, as well as the decompiled scripts as reference.
